src/rgw/rgw_auth_s3.cc

```diff
@@ -457,7 +457,7 @@ int parse_credentials(const req_info& info,                     /* in */
   return 0;
 }
 
-static inline bool char_needs_aws4_escaping(const char c)
+bool char_needs_aws4_escaping(const char c, bool encodeSlash)
```

Comment:
this change should not remove static and inline specifiers; encodeSlash is ok, but it should to stick with local convention be "encode_slash" probably